packages/context/src/__tests__/unit/binding.unit.ts

```diff
@@ -237,6 +238,51 @@ describe('Binding', () => {
       expect(value).to.eql('Hello, John');
     });
 
+    it('supports a factory to use async static provider to inject binding', async () => {
+      class GreetingProvider {
+        static async value(
+          @inject('user') user: string,
+          @inject.binding() currentBinding: Binding,
```

Comment:
A question: how is the binding/config injection related to the `options.session` in the code change?

Reply:
Such injections rely on the current binding from `options.session`.